File: framework/src/play-jcache/src/main/scala/play/api/libs/jcache/JCacheComponents.scala
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.libs.jcache

import javax.cache.{ CacheManager, Caching }

import play.api.Environment

/**
 * Components for JCache CacheManager
 */
trait JCacheComponents {

  def environment: Environment

  lazy val cacheManager: CacheManager = Caching.getCachingProvider(environment.classLoader).getCacheManager
}

File: framework/src/play-jcache/src/main/java/play/libs/jcache/JCacheComponents.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.libs.jcache;

import play.Environment;

import javax.cache.CacheManager;
import javax.cache.Caching;

/**
 * JCache components
 */
public interface JCacheComponents {

    Environment environment();

    default CacheManager cacheManager() {
        return Caching.getCachingProvider(environment().classLoader()).getCacheManager();
    }

}
